src/app/providers/theme-provider/ThemeProvider.jsx

```diff
@@ -0,0 +1,23 @@
+import { useState } from 'react';
+import { ThemeProvider as StyleComponentThemeProvider } from 'styled-components';
+import { lightTheme, darkTheme } from 'theme';
+import { THEME_STORAGE_NAME } from 'constants/localStorage';
+import { DARK, LIGHT } from 'constants/theme';
+import { ThemeContext } from './duck';
+
+const ThemeProvider = ({ children }) => {
+  const [themeName, setThemeName] = useState(
```

Comment:
I guess you are not yet using your useChangeTheme hook, but I wonder if you could use something like useLocalStorageValue hook that I was showing on our first lecture
```
const [themeName, setThemeName] = useLocalStorageValue(THEME_STORAGE_NAME, LIGHT);
```

It would incapsulate both getting initial value and managing setting it to local storage whenever it was changed

what do you think?